app/scripts/services/rest.js

```diff
@@ -57,4 +52,35 @@ angular.module('openhimWebui2App')
       }),
 
     };
+  })
+  .factory('Import', function ($rootScope, $resource) {
+
+    // fetch API server details
+    var protocol = angular.copy( $rootScope.protocol );
+    var host = angular.copy( $rootScope.host );
+    var port = angular.copy( $rootScope.port );
+    var server = protocol + '://' + host + ':' + port;
+
+    return {
+      Channels: $resource( server + '/channels/:channelId', { channelId: '@_id' }, {
+        update: { method: 'PUT' }
+      }),
+
+      Users: $resource( server + '/users/:userId', { userId: '@_id' }, {
+        update: { method: 'PUT' }
+      }),
+
+      Clients: $resource( server + '/clients/:clientId/', { clientId: '@_id' }, {
+        update: { method: 'PUT' }
+      }),
+
+      Mediators: $resource( server + '/mediators:mediatorId/', { mediatorId: '@_id' }, {
+        update: { method: 'PUT' }
+      }),
+
+      ContactGroups: $resource( server + '/groups/:groupId', { groupId: '@_id' }, {
+        update: { method: 'PUT' }
+      }),
```

Comment:
Ok I see. I'd prefer to keep this to one API but if that's not possible then I think you have a good solution here.